dotCMS/src/main/java/com/dotcms/rest/api/v1/content/search/strategies/FieldStrategyFactory.java

```diff
@@ -0,0 +1,65 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This Factory class is responsible for loading the different Field Strategies that generate the
+ * Lucene queries. System Field Strategies represent Lucene terms that do not belong to Content
+ * Type fields, whereas User Searchable Field Strategies represent the Lucene terms that belong to
+ * actual fields in a Contentlet.
+ *
+ * @author Jose Castro
+ * @since Jan 29th, 2025
+ */
+public class FieldStrategyFactory {
+
+    private static final Map<FieldHandlerId, FieldStrategy> SYSTEM_FIELD_STRATEGY_MAP = new HashMap<>();
+    private static final Map<FieldHandlerId, FieldStrategy> SEARCHABLE_FIELD_STRATEGY_MAP = new HashMap<>();
+
+    static {
+        // Here, we're associating each Field Handler with its corresponding Field Strategy for the
+        // System Searchable fields/attributes. It's worth noting that These are NOT actual Content
+        // Type fields per se. But, they allow users to query contents via Lucene
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.CONTENT_TYPE_IDS, new ContentTypesFieldStrategy());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.SITE_ID, new SiteAttributeStrategy());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.GLOBAL_SEARCH, new GlobalSearchAttributeStrategy());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.VARIANT, new VariantAttributeStrategy());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.LANGUAGE, new LanguageAttributeStrategy());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.WORKFLOW_SCHEME, new WorkflowSchemeAttributeAction());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.WORKFLOW_STEP, new WorkflowStepAttributeAction());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.ARCHIVED_CONTENT, new ArchivedContentAttributeStrategy());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.LOCKED_CONTENT, new LockedContentAttributeStrategy());
+        SYSTEM_FIELD_STRATEGY_MAP.put(FieldHandlerId.LIVE_CONTENT, new UnpublishedContentAttributeStrategy());
+
+        // Here, we're associating each Field Handler with its corresponding Field Strategy for the
+        // group of actual Content Type fields that share the same query formatting, escaping and
+        // processing for their specific values
+        SEARCHABLE_FIELD_STRATEGY_MAP.put(FieldHandlerId.TEXT, new TextFieldStrategy());
+        SEARCHABLE_FIELD_STRATEGY_MAP.put(FieldHandlerId.BINARY, new BinaryFieldStrategy());
+        SEARCHABLE_FIELD_STRATEGY_MAP.put(FieldHandlerId.DATE_TIME, new DateTimeFieldStrategy());
+        SEARCHABLE_FIELD_STRATEGY_MAP.put(FieldHandlerId.KEY_VALUE, new KeyValueFieldStrategy());
+        SEARCHABLE_FIELD_STRATEGY_MAP.put(FieldHandlerId.CATEGORY, new CategoryFieldStrategy());
+        SEARCHABLE_FIELD_STRATEGY_MAP.put(FieldHandlerId.RELATIONSHIP, new RelationshipFieldStrategy());
+        SEARCHABLE_FIELD_STRATEGY_MAP.put(FieldHandlerId.TAG, new TagFieldStrategy());
+    }
+
```

Comment:
it would be great to add some kind of extensibility to the map
if it is not needed, considerer making immutable map instead

Reply:
Done!